api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3794,12 +3800,126 @@ public void saveOrderGroup_shouldFailValidationIfAnyOrdersFailValidation() {
 		Exception expectedGroupValidationError = null;
 		try {
 			Context.getOrderService().saveOrderGroup(orderGroup);
-		}
-		catch (Exception e) {
+		} catch (Exception e) {
 			expectedGroupValidationError = e;
 		}
 
 		assertNotNull(expectedGroupValidationError, "Validation should cause order group to fail to save");
 		assertEquals(expectedValidationError.getMessage(), expectedGroupValidationError.getMessage());
 	}
+	
+	@Test
+	public void getOrderGroupAttributeTypes_shouldReturnAllOrderGroupAttributeTypes(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		List<OrderGroupAttributeType>orderGroupAttributeTypes = orderService.getAllOrderGroupAttributeTypes();
+		assertEquals(4, orderGroupAttributeTypes.size());
+	}
+	
+	@Test
+	public void getOrderGroupAttributeType_shouldReturnNullIfNoOrderGroupAttributeTypeHasTheGivenId(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
```

Comment:
Instead of doing executeDataSet in each test method, isn't it better to do it once with a BeforeClass?

Reply:
I had actually opted for that , but it was the convention already used in the class , so i thought otherwise.